docs/Developers/Development_Setup.rst

```diff
@@ -11,10 +11,46 @@ Virtualenv
 
 We highly recommend installing coala in a virtualenv for development. This
 will allow you to have a contained environment in which to modify coala,
-separate from any other installation of coala that you may not want to
-break. For more information about virtualenvs, please refer to the
-`virtualenv setup <https://docs.coala.io/en/latest/Help/MAC_Hints.html#create-virtual-environments-with-pyvenv>`__ section for information on setting one
-up.
+separate from any other installation of coala that you may not want to break.
+Here we will be showing how to have a virtualenv using :code:`venv` and
+:code:`virtualenv`.
+
+Using venv
+~~~~~~~~~~
+
+- Make sure to have Python 3 installed in your local machine.
+
+- Setting up virtualenv with venv :
+  ::
+
+      $ python3 -m venv venv
+      # This creates a isolated python environment called venv
+      # in your current directory.
+      # To activate the environment type:
+      $ source venv/bin/activate
+      # To exit the environment simply type:
+      (venv)$ deactivate
+
+- Now you can activate the environment and start installing from git.
+
+Using virtualenv
+~~~~~~~~~~~~~~~~
+
+- Install virtualenv using pip :
+  ::
+      $ pip install virtualenv
+
+- Create the virtualenv :
+  ::
+      $ cd working_dir # move into the dir where you want to create venv
+      $ virtualenv venv
+
+- Run venv :
+  ::
+      $ source venv/bin/activate
+      (venv)$ deactivate # to exit the environment
+
+- After this, you can start installing from git.
```

Comment:
IMO user would be confused over `you can start installing from git`. Link `installing from git`  to its section ?
Its upto you.. :)